src/AzureStack/Admin/FabricAdmin/Fabric.Admin.Tests/src/DriveTests.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+//
+
+namespace Fabric.Tests
+{
+    using Microsoft.AzureStack.Management.Fabric.Admin;
+    using Microsoft.AzureStack.Management.Fabric.Admin.Models;
+    using Xunit;
+
+    public class DriveTests : FabricTestBase
+    {
+
+        private void AssertDrivesAreSame(Drive expected, Drive found)
+        {
+            if (expected == null)
+            {
+                Assert.Null(found);
+            }
+            else
+            {
+                Assert.True(FabricCommon.ResourceAreSame(expected, found));
+
+                Assert.Equal(expected.StorageNode, found.StorageNode);
+                Assert.Equal(expected.SerialNumber, found.SerialNumber);
+                Assert.Equal(expected.HealthStatus, found.HealthStatus);
+                Assert.Equal(expected.OperationalStatus, found.OperationalStatus);
+                Assert.Equal(expected.Usage, found.Usage);
+                Assert.Equal(expected.CanPool, found.CanPool);
+                Assert.Equal(expected.CannotPoolReason, found.CannotPoolReason);
+                Assert.Equal(expected.PhysicalLocation, found.PhysicalLocation);
+                Assert.Equal(expected.Model, found.Model);
+                Assert.Equal(expected.MediaType, found.MediaType);
+                Assert.Equal(expected.CapacityGB, found.CapacityGB);
+                Assert.Equal(expected.Description, found.Description);
+                Assert.Equal(expected.Action, found.Action);
+            }
+        }
+
+        private void ValidateDrive(Drive instance)
+        {
+            FabricCommon.ValidateResource(instance);
+
+            Assert.NotNull(instance.StorageNode);
+            Assert.NotNull(instance.SerialNumber);
+            Assert.NotNull(instance.HealthStatus);
+            Assert.NotNull(instance.OperationalStatus);
+            Assert.NotNull(instance.Usage);
+            Assert.NotNull(instance.CanPool);
+            Assert.NotNull(instance.CannotPoolReason);
+            Assert.NotNull(instance.PhysicalLocation);
+            Assert.NotNull(instance.Model);
+            Assert.NotNull(instance.MediaType);
+            Assert.NotNull(instance.CapacityGB);
+            Assert.NotNull(instance.Description);
+            Assert.NotNull(instance.Action);
+        }
+
+
+        [Fact]
+        public void TestListDrives()
+        {
+            RunTest((client) => {
+                OverStorageSubSystems(client, (fabricLocationName, scaleUnitName, storageSubSystemName) => {
+                    var drives = client.Drives.List(ResourceGroupName, fabricLocationName, scaleUnitName, storageSubSystemName);
+                    Common.MapOverIPage(drives, client.Drives.ListNext, ValidateDrive);
+                    Common.WriteIPagesToFile(drives, client.Drives.ListNext, "ListDrives.txt", ResourceName);
```

Comment:
This method was really only for debugging purposes.  Mainly so I could visually verify everything was okay.  I should do a PR to remove these one day ha ha.